engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/AbstractScriptAssertValidator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv;
+
+import java.lang.annotation.Annotation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.scripting.ScriptEvaluator;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
+import org.hibernate.validator.internal.util.logging.Log;
+import org.hibernate.validator.internal.util.logging.LoggerFactory;
+import org.hibernate.validator.scripting.ScriptEvaluatorNotFoundException;
+
+/**
+ * @author Marko Bekhta
+ */
+public abstract class AbstractScriptAssertValidator<A extends Annotation, T> implements ConstraintValidator<A, T> {
+
+	private static final Log log = LoggerFactory.make();
+
+	protected String languageName;
+	protected String script;
+	protected String escapedScript;
+	protected volatile ScriptAssertContext scriptAssertContext;
+
+	protected ScriptAssertContext getScriptAssertContext(ConstraintValidatorContext constraintValidatorContext) {
+		if ( scriptAssertContext == null ) {
```

Comment:
Why is it done lazily upon `isValid()` and not via `initialize()`?

Reply:
That's actually that problem that @gsmet mentioned in another PR about having some sort of context in initialize method... As the factory is accessible only via `constraintValidatorContext` and there's no way of getting it during init, as we only have an annotation there, there's this lazy initialization ...